src/components/common/UUIDTag.js

```diff
@@ -0,0 +1,12 @@
+import { Box } from '@mui/material';
+
+function UUIDTag({ uuid, sx }) {
+  const formattedId = `${uuid.slice(0, 4)}...${uuid.slice(
+    uuid.length - 4,
+    uuid.length,
+  )}`;
+
+  return <Box sx={sx}>Token #{formattedId}</Box>;
```

Comment:
@RubenSmn great you are working on this, I have some suggestion:

1. `Token #` shouldn't be the knowledge of this component, so we can reuse it somewhere, say: the tree uuid (the tree id would be in form of uuid in the future) 

2. I need a tooltip to display the entire uuid by hovering the mouse. 
3. I need the tooltip to be interactable, so I can cope the id ( for example, a button/link on the tooltip saying: `copy`
4. Use `span` as the tag of this component, so we can embedded it in to the context without breaking the line and we can inherite the style, ( I remember we can indicate the tag for box and so on, like: `<Box component='span' />`, ideally I hope we can simply bring in the function (shorten the id) and do not invade the style.